### PR TITLE
Generic frontend tag search

### PR DESCRIPTION
**Pull request by abocquet**
Implement quickly a sidebar to filter objects by tag using a simple hook

image

+ fix  unconsistent ordering warnings

### CODE REVIEW

File: backend/tags/filters/taggable.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class TaggableFilter(filters.FilterSet):
+    tags__are = django_filters.CharFilter(method="tags_are")
+
+    def tags_are(self, queryset, value, *args, **kwargs):
+        try:
+            if args:
+                tag_ids = args[0].split(",")
+                for tag_id in tag_ids:
+                    new_condition = Q(**{f"tags__id": tag_id})
```
[the remark this answers is not in the document]
Comment (abocquet) [reply backend/tags/filters/taggable.py:22]: f

File: frontend/src/components/utils/sidebar/CheckboxField.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { FormCheck } from "react-bootstrap";
+
+export const CheckboxField = ({ label, state, setState }) => (
```
[the remark this answers is not in the document]
Comment (abocquet) [reply frontend/src/components/utils/sidebar/CheckboxField.tsx:4]: This component is proper to the sidebar

File: backend/tags/filters/taggable.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import django_filters
+from django.db.models import Count, Q
+from django_filters import rest_framework as filters
+
+from tags.models import Tag
+
+"""
+Taggable filter allows to filter taggable objects with an intersection of tags. If you're working with Media for 
+instance, /associations/medias/?tag_id__in=1,2 will return all medias that have a tag with id 1 or 2.
+If you want to query all medias with tag 1 AND 2, use /associations/medias/?tag__are=1,2
```
[the remark this answers is not in the document]
Comment (abocquet) [reply backend/tags/filters/taggable.py:10]: Using `tag__in` would overload the default behavior (which is the "or")

File: backend/tags/filters/taggable.py [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
 from django_filters import rest_framework as filters
 
-from tags.models import Tag
-
 """
-Taggable filter allows to filter taggable objects with an intersection of tags. If you're working with Media for 
-instance, /associations/medias/?tag_id__in=1,2 will return all medias that have a tag with id 1 or 2.
-If you want to query all medias with tag 1 AND 2, use /associations/medias/?tag__are=1,2
+Taggable filter allows to filter taggable objects with an intersection of tags. If you're working with Media 
+for instance, `/associations/medias/?tag__are=1,2` will query all medias with tag 1 AND 2.
```
Comment (Piwakk) [inline backend/tags/filters/taggable.py:7]: Is it `tags` ?
Also, I find it a bit confusing that it's tag__in (no plural) and tags__are (plural)

File: frontend/src/components/utils/sidebar/OptionSidebar.tsx [hunk 5]
```diff
@@ -1,25 +1,43 @@
...
+ * interface SidebarOption {
+ *   notifyChange: callback when the inputs change;
+ *   searchable?: display a search field on the top to look for specific field.
+ *                The search is based on their label
+ *   sections: {
+ *       title: the name of the section
+ *       id: an id (used for React `key` propp)
+ *       retractable: if True, the user will be able to hide the section
+ *       props?: additional properties to be given to the section html parent
+ *       fields: The fields in the section. Currenlty only
```
Comment (Piwakk) [inline frontend/src/components/utils/sidebar/OptionSidebar.tsx:20]: Currently*

File: frontend/src/components/utils/sidebar/OptionSidebar.tsx [hunk 6]
```diff
@@ -1,25 +1,43 @@
...
-import Form from "react-bootstrap/Form";
-import { SidebarOption, SidebarOptionField } from "./interfaces";
-import { CheckboxField } from "./CheckboxField";
+import { SidebarOption } from "./interfaces";
 import { InputField } from "./InputField";
 import Fuse from "fuse.js";
 import { SidebarSection } from "./SidebarSection";
 
+/**
+ * Component for displaying additional options on the right of a page such as
```
Comment (Piwakk) [inline frontend/src/components/utils/sidebar/OptionSidebar.tsx:8]: On the left?

Comment (Piwakk) [reply frontend/src/components/utils/sidebar/OptionSidebar.tsx:8]: Or, you could just say "in the sidebar"

**Pull request merged**